src/game/market.rs

```diff
@@ -60,94 +59,120 @@ impl OrderType {
-    order_type: OrderType,
-    resource_type: String,
-    room_name: RoomName,
-    amount: u32,
-    remaining_amount: u32,
-    price: f64,
+    pub order_type: OrderType,
+    pub resource_type: MarketResourceType,
+    /// Room that owns the order, `None` for intershard orders
+    pub room_name: Option<RoomName>,
+    pub amount: u32,
+    pub remaining_amount: u32,
+    pub price: f64,
 }
 js_deserializable!(Order);
 
 #[derive(Deserialize, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct MyOrder {
-    id: String,
-    created: u32,
-    active: bool,
+    pub id: String,
+    /// Tick of order creation, `None` for intershard orders
+    pub created: Option<u32>,
+    /// Timestamp of order creation in milliseconds since epoch
+    pub created_timestamp: u64,
+    pub active: bool,
     #[serde(rename = "type", deserialize_with = "OrderType::deserialize_from_str")]
-    order_type: OrderType,
-    resource_type: String,
-    room_name: RoomName,
-    amount: u32,
-    remaining_amount: u32,
-    total_amount: u32,
-    price: f64,
+    pub order_type: OrderType,
+    pub resource_type: MarketResourceType,
+    /// Room that owns the order, `None` for intershard orders
+    pub room_name: Option<RoomName>,
+    pub amount: u32,
+    pub remaining_amount: u32,
+    pub total_amount: u32,
+    pub price: f64,
 }
 js_deserializable!(MyOrder);
 
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct OrderHistoryRecord {
+    pub resource_type: MarketResourceType,
+    /// Calendar date in string format, eg "2018-12-31"
+    pub date: String,
+    /// Total number of transactions for this resource on this day
+    pub transactions: u32,
+    /// Total volume of this resource bought and sold on this day
+    pub volume: u32,
+    pub avg_price: f64,
+    pub stddev_price: f64,
+}
+js_deserializable!(OrderHistoryRecord);
+
 pub fn credits() -> f64 {
     js_unwrap!(Game.market.credits)
 }
 
 pub fn incoming_transactions() -> Vec<Transaction> {
-    let arr_transaction_value = js! {
-        return Game.market.incomingTransactions;
-    };
-    arr_transaction_value.try_into().unwrap()
+    js_unwrap!(Game.market.incomingTransactions)
 }
 
 pub fn outgoing_transactions() -> Vec<Transaction> {
-    let arr_transaction_value = js! {
-        return Game.market.outgoingTransactions;
-    };
-    arr_transaction_value.try_into().unwrap()
+    js_unwrap!(Game.market.outgoingTransactions)
 }
 
+/// Get a `HashMap` of the player's currently-listed market orders
 pub fn orders() -> HashMap<String, MyOrder> {
-    let order_book_value = js! {
-        return Game.market.orders;
-    };
-    order_book_value.try_into().unwrap()
+    js_unwrap!(Game.market.orders)
 }
 
-pub fn calc_transaction_cost(amount: u32, room1: &Room, room2: &Room) -> f64 {
-    js_unwrap!(Game.market.calcTransactionCost(@{amount}, @{room1.name()}, @{room2.name()}))
+pub fn calc_transaction_cost(amount: u32, room1: &RoomName, room2: &RoomName) -> f64 {
```

Comment:
Since `RoomName` is the same size as a `u16`, and copy, it should be more efficient to take them by value here and below?

Basing this off the logic from the trivial-copy-pass-by-ref clippy lint: https://rust-lang.github.io/rust-clippy/master/#trivially_copy_pass_by_ref